1.6Funkcje/1.6Funkcje/Program.cs

```diff
@@ -0,0 +1,129 @@
+﻿ChooseTask();
+
+static void ChooseTask()
+{
+    Console.WriteLine("Spotkanie Firmowe - 1");
+    Console.WriteLine("Rozmowa z nativespeakerem - 2");
+    Console.WriteLine("Praca nad projektem - 3");
+
+    var taskNumber = int.Parse(Console.ReadLine());
+
+    Console.Clear();
+
+    ChooseLogMethods(taskNumber);
+}
+
+static TimeSpan WorkLog(int taskNumber, int numberOfMinuts, int numberOfHours = 0, int numberOfDays = 0)
+{
+    Console.Clear();
+    Console.WriteLine("----Zalogowano zadanie!----");
+    Console.WriteLine($"numer zadania: {taskNumber}");
+
+    if (numberOfDays != 0)
+    {
+        Console.WriteLine($"liczba dni: {numberOfDays}");
+    }
+    if(numberOfHours != 0)
+    {
+        Console.WriteLine($"liczba godzin: {numberOfHours}");
+    }
+
+    Console.WriteLine($"liczba minut: {numberOfMinuts}");
+
+    return new TimeSpan(numberOfDays, numberOfHours, numberOfMinuts, 0);
+}
+
+static void ChooseLogMethods(int taskNumber)
+{
+    Console.WriteLine("Jakie dane chcesz podać?");
+    Console.WriteLine("liczba dni, liczba godzin, liczba minut - 1");
+    Console.WriteLine("liczba godzin, liczba minut - 2");
+    Console.WriteLine("liczba minut - 3");
+
+    int methodNumber;
+
+    var success = int.TryParse(Console.ReadLine(), out methodNumber);
+
+    if (!success)
+        ChooseTask();
+    
+
+    if (methodNumber == 1)
+    {
+        LogDaysHoursAndMinutes(taskNumber);
+    }
+    else if (methodNumber == 2)
+    {
+        LogHoursAndMinutes(taskNumber);
+    }
+    else
+    {
+        LogMinutes(taskNumber);
+    }
+    GoBackToMenu(taskNumber);
+}
+
+static void LogDaysHoursAndMinutes(int taskNumber)
+{
+    Console.Write("Podaj liczbę dni: ");
+    int numberOfDays;
+    var dayParsingSuccess = int.TryParse(Console.ReadLine(), out numberOfDays);
+
+    Console.Write("Podaj liczbę godzin: ");
+    int numberOfHours;
+    var hoursParsingSuccess = int.TryParse(Console.ReadLine(), out numberOfHours);
+
+    Console.Write("Podaj liczbę minut: ");
+    int numberOfMinutes;
+    var minutesParsingSuccess = int.TryParse(Console.ReadLine(), out numberOfMinutes);
+
+    if (!dayParsingSuccess || !hoursParsingSuccess || !minutesParsingSuccess)
+    {
+        Console.Write($"Nie udało się zalogować zadania numer {taskNumber}");
+        GoBackToMenu(taskNumber);
```

Comment:
FIX: Musisz wyjść z rekurencji.
```suggestion
        GoBackToMenu(taskNumber);
        return;
```

Reply:
Poprawiłem w poprzednim commicie

Reply:
Napisz w jakim

Reply:
[6f98e4d](https://github.com/broniuu/Wst-pDoProgramowania/pull/1/commits/6f98e4d5bced158fc5a2f5de8e3e3f33d737cca9)